tpls/desul/include/desul/atomics/Fetch_Op_OpenACC.hpp

```diff
@@ -0,0 +1,491 @@
+/*
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+#ifndef DESUL_ATOMICS_FETCH_OP_OPENACC_HPP_
+#define DESUL_ATOMICS_FETCH_OP_OPENACC_HPP_
+
+#include <desul/atomics/Common.hpp>
+#include <algorithm>
+
+#ifdef KOKKOS_COMPILER_NVHPC
+#ifndef DESUL_CUDA_ARCH_IS_PRE_PASCAL
+template <class T, class R>
+using acc_enable_if_supported_arithmetic_type =  std::enable_if_t<std::is_same_v<T,int> \
+	|| std::is_same_v<T,unsigned int> || std::is_same_v<T,unsigned long long> \
+	|| std::is_same_v<T,float> || std::is_same_v<T,double>, R>;
+#else
+template <class T, class R>
+using acc_enable_if_supported_arithmetic_type = std::enable_if_t<std::is_same_v<T,int> \
+	|| std::is_same_v<T,unsigned int> || std::is_same_v<T,unsigned long long> \
+	|| std::is_same_v<T,float>, R>;
+#endif
+#else
+template <class T, class R>
+using acc_enable_if_supported_arithmetic_type = std::enable_if_t<std::is_arithmetic<T>::value, R>;
+#endif
+
+namespace desul {
+namespace Impl {
+
+// clang-format off
+//<editor-fold desc="device_atomic_fetch_{add,sub,mul,div,lshift,rshift,mod,max,min,and,or,xor}">
+#pragma acc routine seq
+template <class T>
+acc_enable_if_supported_arithmetic_type<T,T> device_atomic_fetch_add(
+T* ptr, const T val, MemoryOrderRelaxed, MemoryScopeDevice) {
+  T tmp;
```

Comment:
Nitpick: For atomic_fetch_op I would prefer `tmp -> old`

Reply:
Renamed the variable `tmp` to `old` in all atomic_fetch_*() functions.